### ISSUE TITLE
registry-creds addon ECR assume role ARN is not optional

### ISSUE THREAD

**Issue by msnelling** [kubernetes/minikube#1757]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT
**Minikube version** (use `minikube version`): v0.21.0

**Environment**:
- **OS** (e.g. from /etc/os-release): Windows 10 Enterprise
- **VM Driver** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep DriverName`): hyperv
- **ISO version** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep -i ISO` or `minikube ssh cat /etc/VERSION`): testing
- **Install tools**:
- **Others**:

**What happened**:
Ran the command `minikube addons configure registry-creds` but when prompted for the *optional* ARN for the AWS role to assume it expects an ARN to be entered.
```
Do you want to enable AWS Elastic Container Registry? [y/n]: y
-- Enter AWS Access Key ID: ****
-- Enter AWS Secret Access Key: ****
-- Enter AWS Region: eu-west-1
-- Enter 12 digit AWS Account ID: ****
-- (Optional) Enter ARN of AWS role to assume:
--Error, please enter a value:
-- (Optional) Enter ARN of AWS role to assume:
--Error, please enter a value:
-- (Optional) Enter ARN of AWS role to assume:
--Error, please enter a value:
-- (Optional) Enter ARN of AWS role to assume:
```
**What you expected to happen**:
I expected to not have to enter an ARN

**How to reproduce it** (as minimally and precisely as possible):
Run the command above and opt to enable the ECR.

**Anything else do we need to know**:

**Comment by hanks** [comment]
also hit this and I create a PR #1754 to fix

**Comment by msnelling** [comment]
I'm not sure whether to file this as another bug but it also looks like the region is not being set correctly, the `awsecr-cred` secret generated by the addon has `https://197653683306.dkr.ecr.us-east-1.amazonaws.com` in the `auths` field in `.dockerconfigjson` whereas I can see that `eu-west-1` was correctly specified and is set in the `registry-creds-ecr` secret.

**Comment by neverfox** [comment]
Fixed by #1754.
